lib/spot_price.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+require_relative './aws_helper'
+require 'datadog/statsd'
+
+class SpotPrice
+  def self.record_metrics()
+    new.record_metrics
+  end
+
+  def record_metrics()
+    # instance types we allow in Spot instance groups.
+    %w(c5.xlarge r5.xlarge m5.xlarge c6i.xlarge r6i.xlarge m6i.xlarge).each do |instance_type|
+      statsd.gauge "spot_price", get_spot_price(instance_type), {:tags => [instance_type]}
+    end
+  end
+
+  def get_spot_price(instance_type)
+    ec2client = AwsHelper.ec2_client
+    resp = ec2client.describe_spot_price_history({
+      start_time: Time.now.utc,
```

Comment:
see [sdk doc](https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/EC2/Client.html#describe_spot_price_history-instance_method).